### PR TITLE
Feat/support layout none in component

### LINKED ISSUE #1496: [Enhancement] 优化 Layout 使得 交互组件不需要参与布局

### PR DESCRIPTION
**Pull request by xuefei1313**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #1496 #1459 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 1]
```diff
@@ -257,6 +257,10 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
         this._relatedAxisComponent = bandAxis;
       }
     }
+    this._relatedAxisComponent as CartesianAxis<any>;
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:260]: 这个是？

File: packages/vchart/src/model/layout-model.ts [hunk 2]
```diff
@@ -47,15 +53,17 @@ export abstract class LayoutModel<T extends IModelSpec> extends BaseModel<T> {
...
+    // if (this._layout) {
+    this.updateLayoutAttribute();
+    const layoutRect = this.getLayoutRect();
+    if (this._forceLayoutTag || !isEqual(this._lastLayoutRect, layoutRect)) {
+      this._lastLayoutRect = { ...layoutRect };
     }
-    super.onLayoutEnd(ctx);
+    this._forceLayoutTag = false;
+    // }
+    this._forceLayoutTag = false;
```
Comment (kkxxkk2019) [inline packages/vchart/src/model/layout-model.ts:66]: 重复了

File: packages/vchart/src/model/layout-model.ts [hunk 3]
```diff
@@ -73,29 +81,34 @@ export abstract class LayoutModel<T extends IModelSpec> extends BaseModel<T> {
...
+      ? this._layout.setLayoutStartPosition(pos)
+      : (this._layoutStartPos = merge(this._layoutStartPos, pos));
   }
   getLayoutRect() {
-    return this._layout.getLayoutRect();
+    return this._layout ? this._layout.getLayoutRect() : this._layoutRect;
   }
   setLayoutRect(rect: Partial<ILayoutRect>, levelMap?: Partial<ILayoutRect>) {
-    return this._layout.setLayoutRect(rect, levelMap);
+    return this._layout ? this._layout.setLayoutRect(rect) : (this._layoutStartPos = merge(this._layoutRect, rect));
```
Comment (kkxxkk2019) [inline packages/vchart/src/model/layout-model.ts:95]: this._layoutStartPos = 不需要给 this._layoutStartPos 赋值吧

Comment (kkxxkk2019) [reply packages/vchart/src/model/layout-model.ts:95]: https://github.com/VisActor/VChart/issues/1459 BUG 记得录入到 case 里哈

File: packages/vchart/src/model/layout-model.ts (hunk 2 above)
Comment (xuefei1313) [reply packages/vchart/src/model/layout-model.ts:66]: done

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts (hunk 1 above)
Comment (xuefei1313) [reply packages/vchart/src/component/data-zoom/data-filter-base-component.ts:260]: 尝试类型转化的，忘删了。。。

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**